runtime/src/comm/ugni/comm-ugni.c

```diff
@@ -350,20 +370,35 @@ static int    registered_heap_info_set;
 static size_t registered_heap_size;
 static void*  registered_heap_start;
 
-#define NUM_MEM_REGIONS 10
+static int    using_hugepages;
+static size_t hugepage_size;
+
+//
+// Memory regions.  mem_regions contains the address/length pairs and
+// uGNI memory domain handles for each memory region of interest to us.
+// mem_region_map contains a copy of every node's mem_regions.
+//
+#define MAX_MEM_REGIONS 100
```

Comment:
Can/should this be env var settable? Obviously this would require doing dynamic allocation instead, but I have no idea how invasive of a change that would be.

I'm ok punting on this for now (and putting it in a post-commit issue) since we've talked about doing full dynamic registration, which might require a different data structure anyways.